src/libraries/SafeERC20Namer.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.20;
+
+import "./AddressStringUtil.sol";
+
+/// @title SafeERC20Namer
+/// @notice produces token descriptors from inconsistent or absent ERC20 symbol implementations that can return string or bytes32
+/// this library will always produce a string symbol to represent the token
+library SafeERC20Namer {
+    function bytes32ToString(bytes32 x) private pure returns (string memory) {
+        bytes memory bytesString = new bytes(32);
+        uint256 charCount = 0;
+        for (uint256 j = 0; j < 32; j++) {
+            bytes1 char = x[j];
+            if (char != 0) {
+                bytesString[charCount] = char;
+                charCount++;
+            }
+        }
+        bytes memory bytesStringTrimmed = new bytes(charCount);
+        for (uint256 j = 0; j < charCount; j++) {
+            bytesStringTrimmed[j] = bytesString[j];
+        }
+        return string(bytesStringTrimmed);
+    }
+
+    /// @notice produces a token symbol from the address - the first 6 hex of the address string in upper case
+    /// @param token the token address
+    /// @return the token symbol
+    function addressToSymbol(address token) private pure returns (string memory) {
+        return AddressStringUtil.toAsciiString(token, 6);
+    }
+
+    /// @notice calls an external view token contract method that returns a symbol, and parses the output into a string
+    /// @param token the token address
+    /// @param selector the selector of the symbol method
+    /// @return the token symbol
+    function callAndParseStringReturn(address token, bytes4 selector) private view returns (string memory) {
+        (bool success, bytes memory data) = token.staticcall(abi.encodeWithSelector(selector));
+        // if not implemented, or returns empty data, return empty string
+        if (!success || data.length == 0) {
```

Comment:
I feel like the second half of this if is unnecessary as its already caught at the end of the function